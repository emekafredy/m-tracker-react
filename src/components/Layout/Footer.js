import React, { Component } from 'react'

class Footer extends Component {
  render() {
    return (
      <footer>
        <p>M-Tracker, Copyright &copy; 2018</p>
      </footer>
    )
  }
}

export default Footer;
