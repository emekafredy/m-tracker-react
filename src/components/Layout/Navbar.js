import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { logUserOut } from '../../actions/auth';

export class Navbar extends Component {

  onLogoutClick(event) {
    event.preventDefault();

    const { logUserOut, history } = this.props;

    logUserOut();
    history.push('/');
  }

  render() {

    const { isAuthenticated, user } = this.props.auth;

    const authMenu = (
      <div className="nav-links">
        <Link to="/new" className="auth-links">
          <i className="fa fa-plus"></i> Create new request
        </Link>
        <Link to="/" onClick={ this.onLogoutClick.bind(this) }
          className="auth-links" id="logoutuser">
          Logout
        </Link>
        <Link to="" className="auth-links"> 
          <span className="role">
            <i className="fa fa-user-circle-o"></i> { isAuthenticated ? user.user.firstname : ''}
          </span>
        </Link>
      </div> 
    );

    const menu = (
      <div className="nav-links">
        <Link to="/register" className="auth-links">Sign Up</Link>
        <Link to="/login" className="auth-links">Login</Link>
      </div> 
    );

    return (
      <div className="navbar">
        <div id="brand">
          <span>
            <Link to="/">M-Tracker</Link>
          </span>
        </div>
        <div className="nav-icon">

        <label htmlFor="nav-check">
          <span></span>
          <span></span>
          <span></span>
        </label>
        </div>

        <input type="checkbox" id="nav-check"/>
        { isAuthenticated ? authMenu : menu }   
      </div>
    );
  }
}

Navbar.propTypes = {
  logUserOut: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
}

const mapStateToProps = (state) => ({
  auth: state.auth
})

export default connect(mapStateToProps, { logUserOut })(withRouter(Navbar));