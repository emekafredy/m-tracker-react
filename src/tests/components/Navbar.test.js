import React from 'react';
import { shallow } from 'enzyme';

import { Navbar } from '../../components/Layout/Navbar';


describe('Navbar', () => {

  test("renders the Navbar component", () => {
    const props = {
      logoutUser: jest.fn(),
      auth: {
        isAuthenticated: true,
        user: {
          user: {
            firstname: 'emeka'
          }
        }
      }
    };

    const wrapper = shallow(<Navbar {...props} />);

    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('should successfully log user out', () => {
    const props = {
      logUserOut: jest.fn(),
      auth: {
        isAuthenticated: true,
        user: {
          user: {
            firstname: 'emeka'
          }
        }
      },
      history: {
        push: jest.fn(),
      }
    };

    const event = {
      preventDefault: jest.fn(),
    }

    const wrapper = shallow(<Navbar {...props} />);

    const logout = wrapper.find('#logoutuser');
    logout.simulate('click', event);

    expect(props.logUserOut).toHaveBeenCalled();
    expect(wrapper.exists()).toBe(true);
  });
})
