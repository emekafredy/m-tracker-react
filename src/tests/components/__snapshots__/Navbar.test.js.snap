// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the Navbar component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "user": Object {
            "firstname": "emeka",
          },
        },
      }
    }
    logoutUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="brand"
        >
          <span>
            <Link
              replace={false}
              to="/"
            >
              M-Tracker
            </Link>
          </span>
        </div>,
        <div
          className="nav-icon"
        >
          <label
            htmlFor="nav-check"
          >
            <span />
            <span />
            <span />
          </label>
        </div>,
        <input
          id="nav-check"
          type="checkbox"
        />,
        <div
          className="nav-links"
        >
          <Link
            className="auth-links"
            replace={false}
            to="/new"
          >
            <i
              className="fa fa-plus"
            />
             Create new request
          </Link>
          <Link
            className="auth-links"
            id="logoutuser"
            onClick={[Function]}
            replace={false}
            to="/"
          >
            Logout
          </Link>
          <Link
            className="auth-links"
            replace={false}
            to=""
          >
            <span
              className="role"
            >
              <i
                className="fa fa-user-circle-o"
              />
               
              emeka
            </span>
          </Link>
        </div>,
      ],
      "className": "navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <Link
              replace={false}
              to="/"
            >
              M-Tracker
            </Link>
          </span>,
          "id": "brand",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              M-Tracker
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "M-Tracker",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "M-Tracker",
            "type": [Function],
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            htmlFor="nav-check"
          >
            <span />
            <span />
            <span />
          </label>,
          "className": "nav-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span />,
              <span />,
              <span />,
            ],
            "htmlFor": "nav-check",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "nav-check",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="auth-links"
              replace={false}
              to="/new"
            >
              <i
                className="fa fa-plus"
              />
               Create new request
            </Link>,
            <Link
              className="auth-links"
              id="logoutuser"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Logout
            </Link>,
            <Link
              className="auth-links"
              replace={false}
              to=""
            >
              <span
                className="role"
              >
                <i
                  className="fa fa-user-circle-o"
                />
                 
                emeka
              </span>
            </Link>,
          ],
          "className": "nav-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-plus"
                />,
                " Create new request",
              ],
              "className": "auth-links",
              "replace": false,
              "to": "/new",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-plus",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Create new request",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logout",
              "className": "auth-links",
              "id": "logoutuser",
              "onClick": [Function],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Logout",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span
                className="role"
              >
                <i
                  className="fa fa-user-circle-o"
                />
                 
                emeka
              </span>,
              "className": "auth-links",
              "replace": false,
              "to": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-user-circle-o"
                  />,
                  " ",
                  "emeka",
                ],
                "className": "role",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-user-circle-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " ",
                "emeka",
              ],
              "type": "span",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="brand"
          >
            <span>
              <Link
                replace={false}
                to="/"
              >
                M-Tracker
              </Link>
            </span>
          </div>,
          <div
            className="nav-icon"
          >
            <label
              htmlFor="nav-check"
            >
              <span />
              <span />
              <span />
            </label>
          </div>,
          <input
            id="nav-check"
            type="checkbox"
          />,
          <div
            className="nav-links"
          >
            <Link
              className="auth-links"
              replace={false}
              to="/new"
            >
              <i
                className="fa fa-plus"
              />
               Create new request
            </Link>
            <Link
              className="auth-links"
              id="logoutuser"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Logout
            </Link>
            <Link
              className="auth-links"
              replace={false}
              to=""
            >
              <span
                className="role"
              >
                <i
                  className="fa fa-user-circle-o"
                />
                 
                emeka
              </span>
            </Link>
          </div>,
        ],
        "className": "navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              <Link
                replace={false}
                to="/"
              >
                M-Tracker
              </Link>
            </span>,
            "id": "brand",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                M-Tracker
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "M-Tracker",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "M-Tracker",
              "type": [Function],
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              htmlFor="nav-check"
            >
              <span />
              <span />
              <span />
            </label>,
            "className": "nav-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span />,
                <span />,
                <span />,
              ],
              "htmlFor": "nav-check",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "nav-check",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="auth-links"
                replace={false}
                to="/new"
              >
                <i
                  className="fa fa-plus"
                />
                 Create new request
              </Link>,
              <Link
                className="auth-links"
                id="logoutuser"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                Logout
              </Link>,
              <Link
                className="auth-links"
                replace={false}
                to=""
              >
                <span
                  className="role"
                >
                  <i
                    className="fa fa-user-circle-o"
                  />
                   
                  emeka
                </span>
              </Link>,
            ],
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-plus"
                  />,
                  " Create new request",
                ],
                "className": "auth-links",
                "replace": false,
                "to": "/new",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Create new request",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Logout",
                "className": "auth-links",
                "id": "logoutuser",
                "onClick": [Function],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Logout",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span
                  className="role"
                >
                  <i
                    className="fa fa-user-circle-o"
                  />
                   
                  emeka
                </span>,
                "className": "auth-links",
                "replace": false,
                "to": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-user-circle-o"
                    />,
                    " ",
                    "emeka",
                  ],
                  "className": "role",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-user-circle-o",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  "emeka",
                ],
                "type": "span",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
